### PR TITLE
feat: adds more environment variables to be used when authenticating with an AWS Role

### PR DESCRIPTION
**Pull request by marcosuma**
Adds two more environment variables ASSUME_ROLE_ARN and SECRET_NAME as values from which it is possible to set a default in the provider configuration (to be used for AWS Role-based authentication).

Related to: https://jira.mongodb.org/browse/INTMDB-1034

## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [X] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [X] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [X] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/fw_provider.go [hunk 1]
```diff
@@ -354,6 +394,13 @@ func setDefaultValuesWithValidations(data *tfMongodbAtlasProviderModel, awsRoleD
 		}, "").(string))
 	}
 
+	if data.SecretName.ValueString() == "" {
```
Comment (lantoli) [inline mongodbatlas/fw_provider.go:395]: do we need yo check if we can't find the secret name?

**Review by lantoli — APPROVED** [review]

File: mongodbatlas/fw_provider.go [hunk 2]
```diff
@@ -70,6 +72,20 @@ type tfAssumeRoleModel struct {
...
+		"transitive_tag_keys": types.SetType{ElemType: types.StringType},
+		"tags":                types.MapType{ElemType: types.StringType},
+		"duration":            types.StringType,
+		"external_id":         types.StringType,
+		"policy":              types.StringType,
+		"role_arn":            types.StringType,
+		"session_name":        types.StringType,
+		"source_identity":     types.StringType,
+	}
+}
```
Comment (AgustinBettati) [inline mongodbatlas/fw_provider.go]:
```suggestion
var AssumeRoleType = types.ObjectType{AttrTypes: map[string]attr.Type{
		"policy_arns":         types.SetType{ElemType: types.StringType},
		"transitive_tag_keys": types.SetType{ElemType: types.StringType},
		"tags":                types.MapType{ElemType: types.StringType},
		"duration":            types.StringType,
		"external_id":         types.StringType,
		"policy":              types.StringType,
		"role_arn":            types.StringType,
		"session_name":        types.StringType,
		"source_identity":     types.StringType,
	}
}
```
Can be directly defined as variable and used in ListValueFrom method

File: mongodbatlas/fw_provider.go [hunk 3]
```diff
@@ -293,6 +308,31 @@ func setDefaultValuesWithValidations(data *tfMongodbAtlasProviderModel, awsRoleD
...
+			data.AssumeRole, diags = types.ListValueFrom(ctx, types.ObjectType{AttrTypes: (&tfAssumeRoleModel{}).AttrTypes()}, []tfAssumeRoleModel{
+				{
+					Tags:              types.MapNull(types.StringType),
+					PolicyARNs:        types.SetNull(types.StringType),
+					TransitiveTagKeys: types.SetNull(types.StringType),
+					RoleARN:           types.StringValue(assumeRoleArn),
+				},
+			})
+			if diags.HasError() {
+				resp.Diagnostics.AddError(diags.Errors()[0].Summary(), diags.Errors()[0].Detail())
```
Comment (AgustinBettati) [inline mongodbatlas/fw_provider.go]:
```suggestion
				resp.Diagnostics.Append(diags...)
```

**Review by AgustinBettati — APPROVED** [review]
small improvements, change LGMT

File: mongodbatlas/fw_provider.go (hunk 1 above)
Comment (marcosuma) [reply mongodbatlas/fw_provider.go:395]: based on what we do with the other values, no

File: mongodbatlas/fw_provider.go (hunk 3 above)
Comment (marcosuma) [reply mongodbatlas/fw_provider.go]: Thank you!

File: mongodbatlas/fw_provider.go (hunk 2 above)
Comment (marcosuma) [reply mongodbatlas/fw_provider.go]: Thank you!

**Review by AgustinBettati — APPROVED** [review]

File: mongodbatlas/provider.go [hunk 4]
```diff
@@ -269,6 +267,8 @@ func providerConfigure(ctx context.Context, d *schema.ResourceData) (any, diag.D
 		RealmBaseURL: d.Get("realm_base_url").(string),
 	}
 
+	assumeRoleValue, ok := d.GetOk("assume_role")
+	awsRoleDefined := (ok && len(assumeRoleValue.([]interface{})) > 0 && assumeRoleValue.([]interface{})[0] != nil)
```
Comment (lantoli) [inline mongodbatlas/provider.go]:
```suggestion
	awsRoleDefined := ok && len(assumeRoleValue.([]any)) > 0 && assumeRoleValue.([]any)[0] != nil
```

File: mongodbatlas/provider.go [hunk 5]
```diff
@@ -308,6 +308,23 @@ func setDefaultsAndValidations(d *schema.ResourceData, awsRoleDefined bool) diag
 		return append(diagnostics, diag.FromErr(err)...)
 	}
 
+	awsRoleDefined := false
+	assumeRoles := d.Get("assume_role").([]interface{})
```
Comment (lantoli) [inline mongodbatlas/provider.go]:
```suggestion
	assumeRoles := d.Get("assume_role").([]any)
```

File: scripts/generate-credentials-with-sts-assume-role.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+}
+# Convert the input string to lowercase
+aws_region=$(to_lowercase "$AWS_REGION")
+# Replace all underscores with hyphens
+aws_region=${aws_region//_/-}
+# e.g. from US_EAST_1 to us-east-1
+
+# Get the STS credentials
+export AWS_REGION="$aws_region"
+CREDENTIALS=$(aws sts assume-role --role-arn "$ASSUME_ROLE_ARN" --role-session-name newSession --output text --query 'Credentials.[AccessKeyId, SecretAccessKey, SessionToken]')
```
Comment (lantoli) [inline scripts/generate-credentials-with-sts-assume-role.sh:21]: do we want to check if aws command fails?

Comment (marcosuma) [reply scripts/generate-credentials-with-sts-assume-role.sh:21]: going to add `set -Eeou pipefail` at the start of the script. How does that sound?

Comment (lantoli) [reply scripts/generate-credentials-with-sts-assume-role.sh:21]: that's great! we should probably use that in all/most scripts.

**Pull request merged**